aws/resource_aws_sfn_state_machine.go

```diff
@@ -121,6 +122,16 @@ func resourceAwsSfnStateMachineRead(d *schema.ResourceData, meta interface{}) er
 	if err := d.Set("creation_date", sm.CreationDate.Format(time.RFC3339)); err != nil {
 		log.Printf("[DEBUG] Error setting creation_date: %s", err)
 	}
+	tagsResp, err := conn.ListTagsForResource(
+		&sfn.ListTagsForResourceInput{
+			ResourceArn: aws.String(d.Id()),
+		},
+	)
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for Step Function: %s. %s", aws.StringValue(sm.Name), err)
```

Comment:
We should always return the error instead of logging a debug message to alert operators that Terraform resource functionality is not working as expected. If there are non-AWS Commercial environments that require special handling because the API does not exist (e.g. "UnsupportedOperation"), we can add that logic with the `err != nil` check. 👍